crates/ruma-common/src/events/room/member/change.rs

```diff
@@ -112,6 +139,11 @@ pub(super) fn membership_change<'a>(
         (St::Join, St::Ban) => Ch::KickedAndBanned,
         (St::Leave, St::Invite) => Ch::Invited,
         (St::Ban, St::Leave) => Ch::Unbanned,
+        (St::Leave, St::Knock) => Ch::Knocked,
+        (St::Invite, St::Knock) => Ch::KnockedAgain,
```

Comment:
The spec says "Must never happen" for this.

Reply:
Not in the unstable spec. This is a new clarification (see #1141).

Reply:
Hm, okay. I think that "knocked again" for invite > knock is a confusing name though, especially given knock > knock is "no change". Maybe we should just use the same "knocked" variant?

Reply:
Sure.